.github/workflows/qc.yml

```diff
@@ -22,13 +22,37 @@ jobs:
         with:
           python-version: ${{ matrix.python-version }}
 
+      #----------------------------------------------
+      #          install & configure poetry
+      #----------------------------------------------
       - name: Install Poetry
-        run: |
-          pip install --upgrade pip
-          pip install poetry==1.3.2
+        uses: snok/install-poetry@v1
+        with:
+          version: 1.3.2
 
+      #----------------------------------------------
+      #       load cached venv if cache exists
+      #----------------------------------------------
+      - name: Load cached venv
```

Comment:
What is the reason for this step? Is there ever going to be a cached environment?